src/state/selectors/axisSelectors.ts

```diff
@@ -894,11 +896,14 @@ export const combineNumericalDomain = (
     return domainFromUserPreference;
   }
 
-  return parseNumericalUserDomain(
-    domainDefinition,
-    mergeDomains(domainOfStackGroups, referenceElementsDomain, computeNumericalDomain(allDataWithErrorDomains)),
-    axisSettings.allowDataOverflow,
-  );
+  const shouldIncludeDomainOfStackGroups =
+    (layout === 'vertical' && axisType === 'xAxis') || (layout === 'horizontal' && axisType === 'yAxis');
+
+  const mergedDomains = shouldIncludeDomainOfStackGroups
```

Comment:
I'm confused, this works with the latest code on main? 

I pulled down the previous PR and this change did not resolve the issue anymore. Weird

Reply:
checked it out and looks good